gtest-parallel

```diff
@@ -414,14 +416,14 @@ for logfile in os.listdir(options.output_dir):
 # Run the specified job. Return the elapsed time in milliseconds if
 # the job succeeds, or None if the job fails. (This ensures that
 # failing tests will run first the next time.)
-def run_job((command, job_id, test)):
+def run_job((command, job_id, test, test_index)):
   begin = time.time()
 
-  with tempfile.NamedTemporaryFile(dir=options.output_dir, delete=False) as log:
+  test_name = re.sub(r'\W', '_', test) + '-' + str(test_index) + '.log'
```

Comment:
The meaning of \W is locale-dependent, and quite complex:

> When the LOCALE and UNICODE flags are not specified, matches any non-alphanumeric character; this is equivalent to the set [^a-zA-Z0-9_]. With LOCALE, it will match any character not in the set [0-9_], and not defined as alphanumeric for the current locale. If UNICODE is set, this will match anything other than [0-9_] plus characters classified as not alphanumeric in the Unicode character properties database.

Maybe just say [^a-zA-Z0-9], which is slightly longer but easier to reason about? It would be bad to let stuff like dots and slashes through here...